src/StoreKeeper/WooCommerce/B2C/Endpoints/TaskProcessor/TaskProcessorEndpoint.php

```diff
@@ -19,16 +19,27 @@ class TaskProcessorEndpoint extends AbstractEndpoint
      */
     public function handle()
     {
+        CronOptions::set(CronOptions::LAST_PRE_EXECUTION_DATE, date('Y-m-d H:i:s'));
+        $beforeCount = ProcessAllTasks::countTasks();
         try {
             $runner = Core::getCommandRunner();
-            $runner->execute(ProcessAllTasks::getCommandName(), [
-                'task_limit' => self::TASK_LIMIT,
+            $runner->execute(ProcessAllTasks::getCommandName(), [], [
+                'limit' => self::TASK_LIMIT,
             ]);
-            StoreKeeperOptions::set(StoreKeeperOptions::CRON_EXECUTION_LAST_STATUS, CronRegistrar::STATUS_SUCCESS);
+            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_SUCCESS);
+            CronOptions::delete(CronOptions::LAST_POST_EXECUTION_ERROR);
         } catch (\Throwable $throwable) {
-            StoreKeeperOptions::set(StoreKeeperOptions::CRON_EXECUTION_LAST_STATUS, CronRegistrar::STATUS_FAILED);
+            CronOptions::set(CronOptions::LAST_EXECUTION_STATUS, CronRegistrar::STATUS_FAILED);
+            CronOptions::set(CronOptions::LAST_POST_EXECUTION_ERROR, $throwable->getMessage());
             $this->logger->error($throwable->getMessage());
             throw new WpRestException($throwable->getMessage(), 401, $throwable);
+        } finally {
+            $afterCount = ProcessAllTasks::countTasks();
+            if ($beforeCount !== $afterCount && 0 !== $beforeCount) {
+                CronOptions::set(CronOptions::LAST_EXECUTION_HAS_PROCESSED, 'true');
+            } else {
+                CronOptions::set(CronOptions::LAST_EXECUTION_HAS_PROCESSED, 'false');
+            }
```

Comment:
i see this 3rd time, please make function for it

Reply:
Added `updateHasProcessed`, as well as `updateFailedExecution` and  `updateSuccessfulExecution`.